test/index.test.ts

```diff
@@ -135,30 +135,53 @@ describe('Server', function () {
       await this.server.listen()
     })
 
-    it('should close all connections', async function () {
-      const spy = sinon.spy()
-      this.server.on('connection', (serverConn: Connection) => {
-        serverConn.on('stream', (stream: DataAndMoneyStream) => {
-          stream.setReceiveMax(100)
-        })
-        serverConn.on('end', spy)
-      })
+    it('safely closes all connections', async function () {
```

Comment:
Were you able to have this fail without your changes? It still passed for me when I removed `pendingRequests` and the timeout.

Reply:
Nice catch!

I updated the test by providing `shouldFulfill` to that server so it waits 1s before fulfilling, and updated the plugin mock so the data handler throws if the plugin is disconnected. Now, I'm able to get that test to fail if I comment out the changes.

Reply:
I'm getting the same test failure as the circleci node v10 build

Reply:
Updated the order of the server close flow to stop handling new requests, await pending requests, close the connection/streams, then disconnect the plugin. Tests are all passing now